### PR TITLE
TECH-646: Create cucumber test for POST /linked-authorization/link-code

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Create cucumber test for POST /linked-authorization/link-code

Ticket: https://govstack-global.atlassian.net/browse/TECH-646

### CODE REVIEW

**Review by mdulko — APPROVED** [review]
lgtm

File: test/openAPI/features/wallet_generate_link_code.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  @negative
+  Scenario: Not able to generate link code because of invalid transaction_id
+    Given Wants to generate link code
+    When Send POST /linked-authorization/link-code request with given X-XSRF-TOKEN header, invalid transactionId and requestTime
+    Then Receive a response from the /linked-authorization/link-code endpoint
+    And The /linked-authorization/link-code endpoint response should be returned in a timely manner 15000ms
+    And The /linked-authorization/link-code endpoint response should have status 200
+    And The /linked-authorization/link-code endpoint response should have content-type: application/json header
+    And The /linked-authorization/link-code endpoint response should match json schema
+    And The /linked-authorization/link-code response should contain errorCode property equals to "invalid_transaction_id"
```
Comment (vishwa-vyom) [inline test/openAPI/features/wallet_generate_link_code.feature:24]: Can we add a negative test case for reuse of a completed transaction id and expecting invalid_transaction_id as error code ?

Comment (vishwa-vyom) [inline test/openAPI/features/wallet_generate_link_code.feature:24]: Can we also have separate test cases for invalid request time and invalid xsrf token id ?

File: test/openAPI/features/wallet_generate_link_code.feature [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@method=POST @endpoint=/linked-authorization/link-code
+Feature: The endpoint to generate link code.
+
+  @smoke @unit @positive
+  Scenario: Successfully generate link code smoke type test
+    Given Wants to generate link code
+    When Send POST /linked-authorization/link-code request with given X-XSRF-TOKEN header, transactionId and requestTime
```
Comment (vishwa-vyom) [inline test/openAPI/features/wallet_generate_link_code.feature:7]: Hope we are also sending the cookie with same value as XSRF-TOKEN

File: test/openAPI/features/wallet_generate_link_code.feature [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    And The /linked-authorization/link-code endpoint response should be returned in a timely manner 15000ms
+    And The /linked-authorization/link-code endpoint response should have status 200
+    And The /linked-authorization/link-code endpoint response should have content-type: application/json header
+    And The /linked-authorization/link-code endpoint response should match json schema
+    And The /linked-authorization/link-code response should contain transactionId property equals provided transactionId
+
+  @negative
+  Scenario: Not able to generate link code because of invalid transaction_id
+    Given Wants to generate link code
+    When Send POST /linked-authorization/link-code request with given X-XSRF-TOKEN header, invalid transactionId and requestTime
```
Comment (vishwa-vyom) [inline test/openAPI/features/wallet_generate_link_code.feature:18]: Here "invalid transactionId" means it could be both blank and a random value ?

**Review by vishwa-vyom — COMMENTED** [review]
Reviewed feature file and added comments.

File: test/openAPI/features/wallet_generate_link_code.feature (hunk 1 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_generate_link_code.feature:24]: Of course, what error code to give for invalid request time and invalid xsrf token id?
In the spec we have now [invalid_transaction_id, invalid_link_code, response_timeout, unknown_error]

File: test/openAPI/features/wallet_generate_link_code.feature (hunk 3 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_generate_link_code.feature:18]: It's for random transactionId. I will add test scenario for blank transactionId

File: test/openAPI/features/wallet_generate_link_code.feature (hunk 1 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_generate_link_code.feature:24]: yes!

Comment (vishwa-vyom) [reply test/openAPI/features/wallet_generate_link_code.feature:24]: invalid request time - we can give invalid_request error code, we have this in the latest spec.
invalid xsrf token - we can assert for a 403 http response code

File: test/openAPI/features/wallet_generate_link_code.feature (hunk 2 above)
Comment (vishwa-vyom) [reply test/openAPI/features/wallet_generate_link_code.feature:7]: this is just a state-less cookie, we don't need any sessions here. Just a new cookie with key as 'XSRF-TOKEN' and value as same given in the X-XSRF-TOKEN request header is enough

File: test/openAPI/features/wallet_generate_link_code.feature (hunk 1 above)
Comment (vishwa-vyom) [reply test/openAPI/features/wallet_generate_link_code.feature:24]: As per #42 the completed transaction id implementation should be that the transaction should be valid, then it has to go through link code, authenticate, consent and link-auth-code. Then the same transaction should be again used for /link-code

Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_generate_link_code.feature:24]: Changes are available in https://github.com/GovStackWorkingGroup/bb-identity/pull/42

Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_generate_link_code.feature:24]: Changes are available in https://github.com/GovStackWorkingGroup/bb-identity/pull/42

File: test/openAPI/features/wallet_generate_link_code.feature (hunk 2 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_generate_link_code.feature:7]: Changes are available in https://github.com/GovStackWorkingGroup/bb-identity/pull/42

Comment (vishwa-vyom) [reply test/openAPI/features/wallet_generate_link_code.feature:7]: cookies key should be 'XSRF-TOKEN' and request header should have 'X-XSRF-TOKEN', there is a small difference in the key

Comment (vishwa-vyom) [reply test/openAPI/features/wallet_generate_link_code.feature:7]: moving comment to new PR and marking this comment as resolved

**Pull request merged**